server/userAuth.js

```diff
@@ -72,6 +68,23 @@ router.use((req, res, next) => {
   res.redirect('/login')
 })
 
+function isAuthorized(user) {
+  const [{value: userEmail = ''} = {}] = user.emails || []
+  const [userDomain] = userEmail.split('@').slice(-1)
+  const checkRegexEmail = (() => {
+    const domainsArray = [...domains]
+    for (const d in domainsArray) {
+      if (userDomain.match(domainsArray[d])) return true
+    }
+  })()
+
+  if ((domains.has(userDomain)) ||
```

Comment:
Could we return the result of this conditional directly? Eg:
```suggestion
return domains.has(userDomain) || domains.has(userEmail) || checkRegexEmail()
```